### ISSUE TITLE
An 'ascii_alphanumerics' variable is missing in the 'strings' lib

### ISSUE THREAD

**Issue by 02744c0c-928c-4b22-a636-10713c5de165** [python/cpython#89832]
BPO | [45669](https://bugs.python.org/issue45669)
--- | :---
Nosy | @smontanaro, @terryjreedy, @serhiy-storchaka, @corona10, @isidentical, @pauloxnet
PRs | <li>python/cpython#29317</li>

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = None
created_at = <Date 2021-10-29.16:11:03.823>
labels = ['type-feature', 'library', '3.11']
title = "An 'ascii_alphanumerics' variable is missing in the 'strings' lib"
updated_at = <Date 2021-11-02.16:52:28.855>
user = 'https://github.com/pauloxnet'
```

bugs.python.org fields:
```python
activity = <Date 2021-11-02.16:52:28.855>
actor = 'terry.reedy'
assignee = 'none'
closed = False
closed_date = None
closer = None
components = ['Library (Lib)']
creation = <Date 2021-10-29.16:11:03.823>
creator = 'pauloxnet'
dependencies = []
files = []
hgrepos = []
issue_num = 45669
keywords = ['patch']
message_count = 9.0
messages = ['405315', '405353', '405359', '405360', '405365', '405368', '405369', '405389', '405517']
nosy_count = 6.0
nosy_names = ['skip.montanaro', 'terry.reedy', 'serhiy.storchaka', 'corona10', 'BTaskaya', 'pauloxnet']
pr_nums = ['29317']
priority = 'normal'
resolution = None
stage = 'patch review'
status = 'open'
superseder = None
type = 'enhancement'
url = 'https://bugs.python.org/issue45669'
versions = ['Python 3.11']
```

</p></details>

**Comment by JelleZijlstra** [comment]
I don't see a strong case for adding this either. It's easy to do this yourself in your own code, and @smontanaro is right that any difference between versions makes it a bit harder for users to write cross-version code.

**Comment by MojoVampire** [comment]
I'm on the -1 team here. I'm not against it for cross-version reasons (it just won't get used in serious projects until support for last version without it is dropped; people writing personal scripts can always use it immediately); I agree with @pauloxnet on that point. If we rejected new stuff because no one could use it for awhile, we'd never do add any public features. I think my metric for an addition of this sort is:

"If we added it, would it be useful enough for people to write compatibility code to get the benefit, and/or upgrade faster to get the benefit?"

which is a proxy for:

"How hard is it to get the same benefits with the current features?"

The answer here would seem to be zero; no one is going to write:

    try:
        from string import ascii_alphanumerics
    except ImportError:
        import string
        ascii_alphanumerics = string.ascii_letters + string.digits

and no one will be the least bit impatient to upgrade to get it; even if they waited, all they'd get was:

    from string import ascii_alphanumerics

instead of the two-liner:

    from string import ascii_letters, digits
    ascii_alphanumerics = ascii_letters + digits

For that matter, when it's not being used for position-insensitive membership testing (e.g. `mystr and all(c in ascii_alphanumerics for c in mystr)` as a bad replacement for `mystr.isascii() and mystr.isalnum()`), the order is going to matter; the name `ascii_alphanumerics` might imply alphabet first (without being clear as to whether lower or upper case stuff comes first; the existing constants provide a precedent for lowercase coming first, even though it's ASCIIbetically after uppercase), but being ASCII focused, it would be equally reasonable to expect it to occur in ASCIIbetical order (which would be digits, then uppercase, then lowercase). The base64 alphabet uses UPPERlowerdigits, but the base85 alphabet goes digitsUPPERlowerPUNCTUATION (the base32 alphabet is even weirder, using UPPER234567 presumably to avoid confusing `0`/`1` with `O`/`I`). For anyone who cares about the order, the odds that they all agree on the *same* order is basically zero; we'd be providing an option useful only to the order-insensitive users and the people who want that specific order, everyone else has to construct it from parts still.

So in short:

1. The benefit is minimal
2. There isn't a single reasonable order to choose that would help everyone using one of these patterns (there are six possible orderings, four of which are all perfectly reasonable orderings [don't have digits in the middle]), so it would just be one more thing for the people using the other three orderings to check, discard, and still have to roll their own.

**Comment by pauloxnet** [comment]
I'd updated the PR to Python 3.13
